pennylane/templates/subroutines/qpe.py

```diff
@@ -0,0 +1,123 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the ``QuantumPhaseEstimation`` template.
+"""
+import pennylane as qml
+from pennylane.templates.decorator import template
+from pennylane.wires import Wires
+
+
+@template
+def QuantumPhaseEstimation(unitary, target_wires, estimation_wires):
+    r"""Performs the
+    `quantum phase estimation <https://en.wikipedia.org/wiki/Quantum_phase_estimation_algorithm>`__
+    circuit.
+
+    Given a unitary matrix :math:`U`, this template applies the circuit for quantum phase
+    estimation. The unitary is applied to the qubits specified by ``target_wires`` and :math:`n`
+    qubits are used for phase estimation as specified by ``estimation_wires``.
+
+    .. figure:: ../../_static/templates/subroutines/qpe.svg
+        :align: center
+        :width: 60%
+        :target: javascript:void(0);
+
+    This circuit can be used to perform the standard quantum phase estimation algorithm, consisting
+    of the following steps:
+
+    #. Prepare ``target_wires`` in an eigenstate of :math:`U`. If that eigenstate has a
+       corresponding eigenvalue :math:`e^{2 \pi i \theta}` with phase :math:`\theta \in [0, 1)`,
+       this algorithm will measure :math:`\theta`.
+    #. Apply the ``QuantumPhaseEstimation`` circuit.
+    #. Measure ``estimation_wires`` using :func:`~.probs`, giving a probability distribution over
+       measurement outcomes in the computational basis.
+    #. Find the index of the largest value in the probability distribution and divide that number by
+       :math:`2^{n}`. This number will be an estimate of :math:`\theta` with an error that decreases
+       exponentially with the number of qubits :math:`n`.
+
+    Note that if :math:`\theta \in (-1, 0]`, we can estimate the phase by again finding the index
+    :math:`i` found in step 4 and calculating :math:`\theta \approx \frac{1 - i}{2^{n}}`. The
+    usage details below give an example of this case.
+
+    Args:
+        unitary (array): the phase estimation unitary, specified as a matrix
+        target_wires (Union[Wires, Sequence[int], or int]): the target wires to apply the unitary
+        estimation_wires (Union[Wires, Sequence[int], or int]): the wires to be used for phase
+            estimation
+
+    Raises:
+        QuantumFunctionError: if the ``target_wires`` and ``estimation_wires`` share a common
+            element
+
+    .. UsageDetails::
+
+        Consider the matrix corresponding to a rotation from an :class:`~.RX` gate:
+
+        .. code-block:: python
+
+            import pennylane as qml
+            from pennylane.templates import QuantumPhaseEstimation
+            from pennylane import numpy as np
+
+            phase = 5
+            target_wires = [0]
+            u = qml.RX(phase, wires=0).matrix
```

Comment:
I know that it's the recommended standard for naming variables, but somehow I find ti disturbing to use `u` instead of `U` to denote a unitary. 

So, minor detail, but maybe use `unitary` instead of `u`?

Reply:
Nice, changed to `unitary`.